### PR TITLE
fix when --out parameter contain more spaces, and support single quotes

### PR DESCRIPTION
**Pull request by weirongxu**

### CODE REVIEW

File: js/ctrls/modal.js [hunk 1]
```diff
@@ -79,15 +79,15 @@ angular
     parse: function() {
       return _
         .chain(this.uris.trim().split(/\r?\n/g))
-        .map(function(d) { 
+        .map(function(d) {
           return _(d)
-            .replace(/("[^"]*")/g, function(c) {
-              return c.replace('%','%25').replace(' ','%20');
+            .replace(/["'][^"]*['"]/g, function(c) {
```
Comment (ziahamza) [inline js/ctrls/modal.js]: Shouldn’t this be /[“‘][^”’][“‘]/g ?

Comment (ziahamza) [reply js/ctrls/modal.js]: @weirongxu

Comment (weirongxu) [reply js/ctrls/modal.js]: Yes, I forget this part
I'll be updated immediately

**Comment by ziahamza** [comment]
Thanks @weirongxu :tada:

**Comment by no1xsyzy** [comment]
Thank you for fixing, I didn't realize that `String#replace` with string instead of regex will only replace once: I uses Python more.

BTW, warning of trap: your way to allow single quotes will also allow unbalanced double/single quote pair, e.g. `"some' 'text"` will still be splitted into two fields, which is different from how shell works.

**Comment by no1xsyzy** [comment]
I will work on more robust way to shell line parsing, and I am afraid it big and common enough to be a standalone library, probably there is already one.

**Pull request merged**